### ISSUE TITLE
Failure generating response

### ISSUE THREAD

**Issue by Emmanuel-Develops** [bitcoinsearch/chat-btc#84]
## problem
I'm not able to get an answer for "silent payments" but other queries work
<img width="200" alt="Screenshot_20240514_032159_Samsung_Internet" src="https://github.com/bitcoinsearch/chat-btc/assets/62019510/04d0bb3f-1788-421c-bbaa-19b19eab629d"><img width="800" alt="Screenshot 2024-05-14 at 12 45 25 PM" src="https://github.com/bitcoinsearch/chat-btc/assets/62019510/24c88977-e929-46a3-ab79-fbdc378ce87e">

## deductions
- cache?: second image shows no cached answers were used
- sources?: second image shows only one source was used.
   Possible explanation: Token limit cuts off the remaining sources as they were too large

**Comment by kouloumos** [comment]
>  Possible explanation: Token limit cuts off the remaining sources as they were too large

Token limit is 7000 characters. [The source that is included](https://bitcoinops.org/en/topics/silent-payments/) ([github](https://github.com/bitcoinops/bitcoinops.github.io/blob/master/_topics/en/silent-payments.md)) is ~273 tokens based on [this tool](https://platform.openai.com/tokenizer). I don't think that this is a token limit.

**Comment by Emmanuel-Develops** [comment]
> edit: Okay, my bad, so what you are saying is that **the remaining sources** add up to more than 7000 tokens. Do we have access to what those sources are? So that we can verify it?

Yeah, but more like the next relevant source in addition with the most relevant (the single source shown) adds up to more than 7000 tokens.

Yeah, on local you can disable the tokenEnforcer and see the sources used or log the sources before then. I ran the query again and these are the results with tokenEnforcer disabled.
Note: the order of the sources may not be exactly what the user got the first time

[[0]: https://bitcoinops.org/en/topics/silent-payments ](https://bitcoinops.org/en/topics/silent-payments)
[[1]: https://btctranscripts.com/bitcoin-explained/silent-payments ](https://btctranscripts.com/bitcoin-explained/silent-payments)
[[2]: https://btctranscripts.com/bitcoin-core-dev-tech/2023-04/2023-04-26-silent-payments ](https://btctranscripts.com/bitcoin-core-dev-tech/2023-04/2023-04-26-silent-payments)
[[3]: https://btctranscripts.com/bitcoin-design/learning-bitcoin-and-design/silent-payments ](https://btctranscripts.com/bitcoin-design/learning-bitcoin-and-design/silent-payments)
[[4]: https://delvingbitcoin.org/t/about-the-wg-silent-payments-category/876 ](https://delvingbitcoin.org/t/about-the-wg-silent-payments-category/876)
[[5]: https://btctranscripts.com/bitcoin-explained/accounts-for-bitcoin-easypaysy ](https://btctranscripts.com/bitcoin-explained/accounts-for-bitcoin-easypaysy)
[[6]: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2022-May/020519.html ](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2022-May/020519.html)
[[7]: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2023-August/021844.html ](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2023-August/021844.html)
[[8]: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2022-August/020885.html ](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2022-August/020885.html)
[[9]: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2023-August/021864.html ](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2023-August/021864.html)

source [1] token count is over 11,000
